xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h

```diff
@@ -0,0 +1,1486 @@
+    }
+    return strReturn;
+  }
+  //----------------------------------------------------------------------------
+  
+  //============================================================================
+  ///
+  /// @ingroup cpp_kodi_vfs
+  /// @brief Return the file name from given complate path string
+  ///
+  /// @param[in] path The complete path include file and directory
+  /// @return Filename from path
+  ///
+  ///
+  /// ------------------------------------------------------------------------
+  ///
+  /// **Example:**
+  /// ~~~~~~~~~~~~~{.cpp}
+  /// #include <kodi/Filesystem.h>
+  /// ...
+  /// std::string fileName = kodi::vfs::GetFileName("special://temp/kodi.log");
+  /// fprintf(stderr, "File name is '%s'\n", fileName.c_str());
+  /// ~~~~~~~~~~~~~
+  ///
+  inline std::string GetFileName(const std::string& path)
+  {
+    /* find the last slash */
+    const size_t slash = path.find_last_of("/\\");
+    return path.substr(slash+1);
+  }
+  //----------------------------------------------------------------------------
+
+  //============================================================================
+  ///
+  /// @ingroup cpp_kodi_vfs
+  /// @brief Return the directory name from given complate path string
+  ///
+  /// @param[in] path The complete path include file and directory
+  /// @return Directory name from path
+  ///
+  ///
+  /// ------------------------------------------------------------------------
+  ///
+  /// **Example:**
+  /// ~~~~~~~~~~~~~{.cpp}
+  /// #include <kodi/Filesystem.h>
+  /// ...
+  /// std::string dirName = kodi::vfs::GetDirectoryName("special://temp/kodi.log");
+  /// fprintf(stderr, "Directory name is '%s'\n", dirName.c_str());
+  /// ~~~~~~~~~~~~~
+  ///
+  inline std::string GetDirectoryName(const std::string& path)
+  {
+    // Will from a full filename return the directory the file resides in.
+    // Keeps the final slash at end and possible |option=foo options.
+
+    size_t iPosSlash = path.find_last_of("/\\");
+    if (iPosSlash == std::string::npos)
+      return ""; // No slash, so no path (ignore any options)
+
+    size_t iPosBar = path.rfind('|');
+    if (iPosBar == std::string::npos)
+      return path.substr(0, iPosSlash + 1); // Only path
+
+    return path.substr(0, iPosSlash + 1) + path.substr(iPosBar); // Path + options
+  }
+  //----------------------------------------------------------------------------
+
+  //============================================================================
+  ///
+  /// @ingroup cpp_kodi_vfs
+  /// @brief Return a size aligned to the chunk size at least as large as the chunk size.
+  ///
+  /// @param[in] chunk The chunk size
+  /// @param[in] minimum The minimum size (or maybe the minimum number of chunks?)
+  /// @return The aligned size
+  ///
+  static inline unsigned int GetChunkSize(unsigned int chunk, unsigned int minimum)
```

Comment:
Some of the global function in the header are static, others are not. What is the criteria? Should be consistent.